src/overflow.rs

```diff
@@ -584,6 +591,25 @@ impl<'a> Context<'a> {
             _ => (),
         }
 
+        match self.force_list_tactic {
+            Some(list_tactic) if list_tactic == ListTactic::Mixed => {
+                if tactic != DefinitiveListTactic::Horizontal {
+                    tactic = DefinitiveListTactic::Mixed
+                }
+            }
+            // If we need to force a tactic (like Vertical), we should only do it if there are
+            // at least 2 items for us to format. Otherwise, use the tactic already determined.
+            Some(list_tactic) if self.items.len() > 1 => {
+                tactic = definitive_tactic(
+                    &*list_items,
+                    list_tactic,
+                    Separator::Comma,
+                    self.one_line_width,
+                )
+            }
+            _ => {}
+        };
```

Comment:
Curious if we couldn't hoist this up to the top of the function with an early return as it seems like many paths for determining the definitive tactic would be known up front. I.e. if the user has set `fn_call_style` to `Vertical` is there really a need to perform all the above analysis and checks?

I know things start to get messy quickly with nesting though so lmk if I'm overlooking something

Reply:
I had a similar thought in mind when I first started working on this. I think all the code leading up to the `match` is important because there's the chance that the last item will be rewritten if it overflows so returning before the rewrite leads to weird formatting. Granted I might have also been getting weird formatting due to the `rewrite_method` issues I described in https://github.com/rust-lang/rustfmt/pull/5337#issuecomment-1122725083. I'll play around with this and see if I can hoist this up earlier in the function